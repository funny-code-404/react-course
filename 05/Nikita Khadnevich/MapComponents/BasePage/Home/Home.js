import React, { Component } from 'react'

export default class Home extends Component {

   render() {
      return (
         <div cl>
            <h3>Главная страница</h3>
         </div>
      )
   }
}
