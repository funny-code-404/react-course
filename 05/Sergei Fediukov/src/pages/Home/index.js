const Home = () => {
    return (
        <div>
            <h1>
                home page
           </h1>
        </div>
    )
}
export default Home