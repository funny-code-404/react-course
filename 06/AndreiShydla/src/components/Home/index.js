import React from "react";

class Home extends React.Component {
  render() {
    return <div>Welcome to Phone Book</div>;
  }
}

export default Home;
