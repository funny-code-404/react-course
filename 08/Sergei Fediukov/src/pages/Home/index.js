import List from '../../components/List'
import FormAdd from '../../components/FormAdd'
import { useState, useEffect, useContext } from 'react'
import { database } from '../../firebase'
import { auth } from '../../firebase';
import { AuthContext } from "../../components/context/Auth.context";

const Home = () => {
    const [data, setData] = useState({})
    useEffect(() => {
        let cleanupFunction = false;
        database().ref('/SET_DATA').on('value', (snapshot) => {
            if (!cleanupFunction) setData(snapshot.val());
        })
        return () => cleanupFunction = true;
    }, [])
    const handleLogOut = () => {
        auth.signOut()
    }
    const { currentUser } = useContext(AuthContext)
    return (
        <div>
            <button onClick={handleLogOut} className='form_btn'>Log out ({currentUser ? currentUser.email : ''})</button>
            <h1>
                Список отелей:
           </h1>
            {Boolean(data) ||
                <p className='warning'>В списке пока нет ни одного отеля</p>
            }
            {Boolean(data) &&
                <List data={data} />
            }
            <FormAdd />
        </div>
    )
}
export default Home