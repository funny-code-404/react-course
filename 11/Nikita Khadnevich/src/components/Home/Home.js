import React from 'react'
import HomePage from '../SmallElems/SmallElems'

const Home = () => {

   return (
      <div className='homePage'>
         <HomePage />
      </div>
   )
}

export default Home;