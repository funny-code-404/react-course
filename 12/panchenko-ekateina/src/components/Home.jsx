import { auth } from "../firebase";
import styled from "styled-components";
import background from "../img/ageOfEmpiresHome.jpg";
import { Container, FullBack } from "./style";

const Title = styled.h2`
  color: #fff;
  font-size: 50px;
  margin-bottom: 20px;
  padding: 100px 0 100px 0;
  text-align: center;
  text-transform: uppercase;

  @media (max-width: 590px) {
    font-size: 30px;
    padding: 100px 0 50px 0;
  }
`;

const Button = styled.button`
  align-items: center;
  background: #a1a38b;
  border-radius: 25px;
  border: 2px solid #fff;
  color: #fff;
  cursor: pointer;
  display: flex;
  font-size: 20px;
  justify-content: center;
  height: 40px;
  margin: auto;
  max-width: 50%;
  min-width: 150px;
  outline: none;
  text-transform: uppercase;
`;

export const Home = () => {
  const handleLogout = () => {
    auth.signOut();
  };

  return (
    <Container background={background}>
      <FullBack>
        <Title>Welcome to Age Of Empires!</Title>
        <Button onClick={handleLogout}>Logout</Button>
      </FullBack>
    </Container>
  );
};
