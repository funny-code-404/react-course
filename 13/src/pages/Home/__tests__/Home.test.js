import React from "react";
import { configure, shallow } from "enzyme";
import Adapter from '@wojtekmaj/enzyme-adapter-react-17';

import Home  from "..";

configure({ adapter: new Adapter() });

describe('PAGES  HOME', () => {

    let wrapper;

    beforeEach(() => {
        wrapper = shallow(<Home />);
    });

    it('to Match Snapshot', () => {
        expect(wrapper).toMatchSnapshot();
    });
    it('component is defined', () => {
        expect(wrapper).toHaveLength(1)
    })
    it('Component is return 1 element', () => {
        expect(wrapper.find('h1')).toHaveLength(1);
    });
});

// npm run test src/pages/Home