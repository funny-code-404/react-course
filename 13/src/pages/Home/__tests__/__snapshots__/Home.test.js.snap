// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`PAGES  HOME to Match Snapshot 1`] = `
<h1>
  Home
</h1>
`;
